provisioner/ansible/provisioner.go

```diff
@@ -648,15 +648,29 @@ func (p *Provisioner) executeGalaxy(ui packer.Ui, comm packer.Communicator) erro
 		collectionArgs = append(collectionArgs, "-p", filepath.ToSlash(p.config.CollectionsPath))
 	}
 
-	roleInstallError := p.invokeGalaxyCommand(roleArgs, ui, comm)
-	// Return the error if the role installation failed before attempting the collection install
-	if roleInstallError != nil {
-		return roleInstallError
+	// Search galaxy_file for roles and collections keywords
+	f, err := ioutil.ReadFile(galaxyFile)
+	if err != nil {
+		return err
+	}
+	hasRoles, _ := regexp.Match(`(?m)^roles:`, f)
+	hasCollections, _ := regexp.Match(`(?m)^collections:`, f)
+
+	// If if roles keyword present (v2 format), or no collections keywork present (v1), install roles
+	if hasRoles || !hasCollections {
```

Comment:
The way I'm reading this, even if neither roles nor collections are defined, we still try to install roles? Is that intentional?

Reply:
Yes, that is intentional. Requirements files in v1 format do not use the `roles` key, as it is implied. If `galaxy_file` is specified, and no `collections` key is included, then we should stick with the old behavior (prior to v1.6.3) of just calling `ansible-galaxy install -r $galaxyFile $args`.

Reply:
aha got it. Thanks. Okay, this lgtm.